src/algebra/ring/aut.lean

```diff
@@ -60,4 +61,34 @@ by refine_struct { to_fun := ring_equiv.to_mul_equiv }; intros; refl
 def to_perm : ring_aut R →* equiv.perm R :=
 by refine_struct { to_fun := ring_equiv.to_equiv }; intros; refl
 
+/-- The tautological action by the group of automorphism of a ring `R` on `R`.-/
+instance apply_mul_semiring_action {R : Type*} [semiring R] :
+mul_semiring_action (ring_aut R) R :=
+{ smul := ($),
+  smul_zero := ring_equiv.map_zero,
+  smul_add := ring_equiv.map_add,
+  smul_one := ring_equiv.map_one,
+  smul_mul := ring_equiv.map_mul,
+  one_smul := λ _, rfl,
+  mul_smul := λ _ _ _, rfl }
+
+@[simp]
+protected lemma
+smul_def {R : Type*} [semiring R] (f : ring_aut R) (r : R) : f • r = f r := rfl
+
+instance apply_has_faithful_smul {R : Type*} [semiring R] : has_faithful_smul (ring_aut R) R :=
+⟨λ _ _, ring_equiv.ext⟩
+
+/-- Each element of the group defines a ring automorphism.
+
+This is a stronger version of `distrib_mul_action.to_add_aut` and
+ `mul_distrib_mul_action.to_mul_aut`. -/
+@[simps]
+def mul_semiring_action.to_ring_aut {G : Type*} {R : Type*} [group G] [semiring R]
```

Comment:
Moving the `: Type*` arguments to a `variables` line will fix any universe linter errors, assuming this is where they're coming from (I can't see on mobile)

Reply:
Thanks, I 've tried something. Let's see if it works

Reply:
Nope, it's still not working..

Reply:
I am not sure what's wrong now. Could you provide some feedback please?